### ISSUE TITLE
[FR] Google Drive integration

### ISSUE THREAD

**Issue by awsomemanNever** [AppFlowy-IO/appflowy#24]
**1~3 main use cases of the proposed feature**
The integration of google drive, so you can directly link your google drive documents inside of appflowy. 

**what types of users can benefit from using your proposed feature**
I would use this feature quite heavily to build a dashboard so i can have direct access to my school books that i have scanned and stored on g drive

**Comment by smaimon** [comment]
The website appflowy.io said "We are committed to protecting your personal information and your right to privacy."

Not sure if any #google integration is commitment to protect our personal information and our right to privacy 🤔

**Comment by smaimon** [comment]
Maybe a solution like Nextcloud or e.foundation could be a commitment to data privacy ?

**Comment by Hari-07** [comment]
> The website appflowy.io said "We are committed to protecting your personal information and your right to privacy."
> 
> Not sure if any #google integration is commitment to protect our personal information and our right to privacy 🤔

I think its more about giving the user the control over their own data. If the user wants the convenience of google drive, it'd probably be a good idea to provide that option

**Comment by giddens9527** [comment]
Yeah, agreed with @Hari-07. It's more about client/user owns the data no matter it's local deployed or third-party services.

**Comment by smaimon** [comment]
I understand and agreed with both of you @Hari-07 & @giddens9527 
It could be in the public interest if all content could be end-to-end encrypted and decrypted by appflowy. This means documents and files are unreadable outside of appflowy if you don't have the key. If even the service administrators do not have access to my information.
We need a privacy by design AppFlowy to ensure protection for the privacy. 
This considerations of privacy issues need to be implemented from the very beginning of the development of the product.

**Comment by rileyhawk1417** [comment]
@annieappflowy should guide(s) be written for various storage options with their specific caveats or google drive is currently still not the best option since there has been so many infrastructure changes?

**Comment by annieappflowy** [comment]
Supported in v0.11.3
